pallets/connectors/src/inbound.rs

```diff
@@ -0,0 +1,275 @@
+		// Should be handled by pallet-foreign-investments
+
+		Ok(())
+	}
+
+	/// Increases an existing redemption order of the investor.
+	///
+	/// Transfers the decreased redemption amount from the holdings of the
+	/// `DomainLocator` account of origination domain of this message into the
+	/// investor account.
+	///
+	/// Assumes that the amount of tranche tokens has been locked in the
+	/// `DomainLocator` account of the origination domain beforehand.
+	pub fn do_increase_redemption(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		amount: <T as Config>::Balance,
+		sending_domain: DomainAddress,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::redemption(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_add(amount)?;
+
+		// Transfer tranche tokens from `DomainLocator` account of origination domain
+		T::Tokens::transfer(
+			invest_id.clone().into(),
+			&DomainLocator::<Domain> {
+				domain: sending_domain.domain(),
+			}
+			.into_account_truncating(),
+			&investor,
+			amount,
+			false,
+		)?;
+
+		T::ForeignInvestment::update_redemption(&investor, invest_id, post_amount)?;
+
+		Ok(())
+	}
+
+	/// Decreases an existing redemption order of the investor.
+	///
+	/// Transfers the decreased redemption amount from the investor account into
+	/// holdings of the `DomainLocator` account of origination domain of this
+	/// message.
+	///
+	/// Initiates a return `ExecutedDecreaseRedemption` message to refund the
+	/// decreased amount on the source domain. The dispatch of this message is
+	/// delayed until the execution of the redemption, e.g. at least until the
+	/// next epoch transition.
+	pub fn do_decrease_redemption(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		amount: <T as Config>::Balance,
+		sending_domain: DomainAddress,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		// NOTE: Required for relaying `ExecutedDecreaseRedemption` message
+		let _currency = Self::try_get_payment_currency(invest_id.clone(), currency_index)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::redemption(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_sub(amount)?;
+
+		T::ForeignInvestment::update_redemption(&investor, invest_id.clone(), post_amount)?;
+
+		// Transfer tranche tokens to `DomainLocator` account of
+		// origination domain
+		T::Tokens::transfer(
+			invest_id.into(),
+			&investor,
+			&DomainLocator::<Domain> {
+				domain: sending_domain.domain(),
+			}
+			.into_account_truncating(),
+			amount,
+			false,
+		)?;
```

Comment:
This must be removed. Same applies as for burning tokens. The moment we trigger the return message is the moment we must do "bookkeeping" for tokens reaching and leaving the chain.